dnas/sensemaker/zomes/coordinator/sensemaker/src/applet.rs

```diff
@@ -10,8 +10,14 @@ use crate::{
     utils::entry_from_record,
 };
 
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct CreateAppletConfigInput {
+    pub applet_config_input: AppletConfigInput,
+    pub role_name: String,
+}
+
 #[hdk_extern]
-pub fn register_applet(applet_config_input: AppletConfigInput) -> ExternResult<AppletConfig> {
+pub fn register_applet(CreateAppletConfigInput { applet_config_input, role_name }: CreateAppletConfigInput) -> ExternResult<AppletConfig> {
```

Comment:
one small comment here, I might have to revisit this again in a separate PR given that one of Damien's review was that the DNA hash (or role_name, whichever is being used in bridge call) where the resource_type came from should be included in the `ResourceType` struct (which might make us not need role name in config since the resource type will have it).

My suggestion is that we merge this PR already and then revisit this when I have the PR for damien's review ready. 😁

Reply:
Sorry @tatssato this slipped my attention and am just seeing this now, but looks like you have gone ahead and merged with #27 !